cmd/derpprobe/derpprobe.go

```diff
@@ -135,13 +138,71 @@ func getOverallStatus() (o overallStatus) {
 	return
 }
 
+func notifySlack(text string) error {
+	type SlackRequestBody struct {
+		Text string `json:"text"`
+	}
+
+	slackBody, err := json.Marshal(SlackRequestBody{Text: text})
+	if err != nil {
+		return err
+	}
+
+	webhookUrl := os.Getenv("SLACK_WEBHOOK")
+	if webhookUrl == "" {
+		return errors.New("No SLACK_WEBHOOK configured")
+	}
+
+	req, err := http.NewRequest(http.MethodPost, webhookUrl, bytes.NewBuffer(slackBody))
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	// if Slack is down, don't break the DERP monitoring. Timeout quickly.
+	client := &http.Client{Timeout: 3 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(resp.Body)
+	if buf.String() != "ok" {
+		return errors.New("Non-ok response returned from Slack")
+	}
+	return nil
+}
+
 func serve(w http.ResponseWriter, r *http.Request) {
 	st := getOverallStatus()
 	summary := "All good"
-	if len(st.bad) > 0 {
+	if (float64(len(st.bad)) / float64(len(st.bad)+len(st.good))) > 0.25 {
+		// This will generate an alert and page a human.
+		// It also ends up in Slack, as part of the alert pipeline.
 		w.WriteHeader(500)
```

Comment:
I thought the plan was to always say 200 OK to Cloudradar and only have it monitor whether derpprobe is running, and then let derpprobe decide what makes it to Slack

Reply:
Right now the only way we have to summon a human is cloudradar initiating a page, so at least for now if enough of the DERP fleet is out to be concerning I think returning 500 is the best way to handle it.